### ISSUE TITLE
In an irreproducible conda environment, tICA outputs all zeros

### ISSUE THREAD

**Issue by rafwiewiora** [markovmodel/PyEMMA#1464]
I have a conda environment that does this, I'm happy to examine it for you however you like, but I can't reproduce this with any new environment. I did waste 3 days getting nonsense results from it though, so it'd be useful to work out what's going on if we can. 

I'm re-running some scoring pipelines on DESRES datasets, had previously done it with 2.5.4 which was just fine, I thought I'd do the re-run with 2.5.7. As we all painfully know, conda gets increasingly slow, so a lazy person like myself might create multiple environments named with different iterations of 'clean'. And so I found myself with a 'clean' environment existing, I can't tell you how or when it came to be, but I updated pyemma in it to 2.5.7, `conda list` is at the end of this post. I run my scoring, and then went back to get top models -- didn't notice anything weird for some proteins, for others I did -- scores too low, timescales too low and unconverged -- so I'm not sure if this happens randomly, or there's some deeper reason why it happens to some systems not others -- only had time to look at one model of villin in particular and _**tICA returns all zeros**_ (and hence kmeans assigns only one state). But clearly this is not the whole story -- I have another model of villin, where I get as many states as I asked, but the timescales are way too low. There's some magic here that depends on the parameters and I've seen conda do weird shit, but not work without errors and give wildly wrong results that depend on parameters. 

Making a fresh environment works just fine, like I said I can't reproduce it, but it really disturbs me that I can have a working setup and get differentially wrong results out of it. Ask me anything. 

```
# Name                    Version                   Build  Channel
_py-xgboost-mutex         2.0                       cpu_0    conda-forge
absl-py                   0.7.1                    py37_0    conda-forge
alabaster                 0.7.12                     py_0    conda-forge
amberlite                 16.0                     pypi_0    pypi
ambertools                17.0                     pypi_0    pypi
apbs                      1.5                  hd8ae8d1_0    conda-forge
asn1crypto                0.24.0                py37_1003    conda-forge
astor                     0.7.1                      py_0    conda-forge
atomicwrites              1.3.0                      py_0    conda-forge
attrs                     19.3.0                     py_0    conda-forge
babel                     2.8.0                      py_0    conda-forge
backcall                  0.1.0                      py_0    conda-forge
bhmm                      0.6.3           py37h3010b51_1002    conda-forge
blas                      2.4                    openblas    conda-forge
blosc                     1.16.3               hf484d3e_0    conda-forge
bzip2                     1.0.6             h14c3975_1002    conda-forge
c-ares                    1.15.0            h14c3975_1001    conda-forge
ca-certificates           2020.4.5.1           hecc5488_0    conda-forge
certifi                   2020.4.5.1       py37hc8dfbb8_0    conda-forge
cffi                      1.12.3           py37h8022711_0    conda-forge
cftime                    1.0.4.2          py37hc1659b7_0    conda-forge
chardet                   3.0.4                 py37_1003    conda-forge
cryptography              2.6.1            py37h72c5cf5_0    conda-forge
curl                      7.65.3               hf8cf82a_0    conda-forge
cycler                    0.10.0                   pypi_0    pypi
cython                    0.29.14          py37he1b5a44_0    conda-forge
dbus                      1.13.2               h714fa37_1  
decorator                 4.4.0                      py_0    conda-forge
dill                      0.2.9                    py37_0    conda-forge
docutils                  0.16                     py37_0    conda-forge
expat                     2.2.5             hf484d3e_1002    conda-forge
fahmunge                  0.3.0                    py37_0    omnia
fftw3f                    3.3.4                         2    omnia
fontconfig                2.13.1            he4413a7_1000    conda-forge
freetype                  2.10.0               he983fc9_0    conda-forge
gast                      0.2.2                      py_0    conda-forge
gettext                   0.19.8.1          h9745a5d_1001    conda-forge
glib                      2.56.2            had28632_1001    conda-forge
grpcio                    1.16.1           py37hf8bcb03_1  
gst-plugins-base          1.14.0               hbbd80ab_1  
gstreamer                 1.14.0               hb453b48_1  
h5py                      2.9.0           nompi_hf008753_1101    conda-forge
hdf4                      4.2.13            hf30be14_1003    conda-forge
hdf5                      1.10.4          nompi_h11e915b_1105    conda-forge
humanfriendly             4.18                     py37_0    conda-forge
icu                       58.2              hf484d3e_1000    conda-forge
idna                      2.8                   py37_1000    conda-forge
imagesize                 1.2.0                      py_0    conda-forge
importlib_metadata        0.23                     py37_0    conda-forge
ipython                   7.5.0            py37h24bf2e0_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
jedi                      0.13.3                   py37_0    conda-forge
jinja2                    2.11.0                     py_0    conda-forge
jpeg                      9c                h14c3975_1001    conda-forge
kiwisolver                1.0.1           py37h6bb024c_1002    conda-forge
krb5                      1.16.3            h05b26f9_1001    conda-forge
latexcodec                2.0.0                      py_0    conda-forge
libblas                   3.8.0                4_openblas    conda-forge
libcblas                  3.8.0                4_openblas    conda-forge
libcurl                   7.65.3               hda55be3_0    conda-forge
libedit                   3.1.20170329      hf8c457e_1001    conda-forge
libffi                    3.2.1             hf484d3e_1005    conda-forge
libgcc-ng                 7.3.0                hdf63c60_0    conda-forge
libgfortran-ng            7.2.0                hdf63c60_3    conda-forge
libiconv                  1.15              h14c3975_1004    conda-forge
liblapack                 3.8.0                4_openblas    conda-forge
liblapacke                3.8.0                4_openblas    conda-forge
libnetcdf                 4.6.2             hbdf4f91_1001    conda-forge
libopenblas               0.3.3                h5a2b251_3  
libpng                    1.6.36            h84994c4_1000    conda-forge
libprotobuf               3.7.1                h8b12597_0    conda-forge
libssh2                   1.8.2                h22169c7_2    conda-forge
libstdcxx-ng              7.3.0                hdf63c60_0    conda-forge
libuuid                   2.32.1            h14c3975_1000    conda-forge
libxcb                    1.13              h14c3975_1002    conda-forge
libxgboost                0.81              hf484d3e_1000    conda-forge
libxml2                   2.9.8             h143f9aa_1005    conda-forge
lzo                       2.10              h14c3975_1000    conda-forge
markdown                  2.6.11                     py_0    conda-forge
markupsafe                1.1.1            py37h516909a_0    conda-forge
matplotlib                3.0.3                    py37_0    conda-forge
matplotlib-base           3.0.3            py37h167e16e_0    conda-forge
mdshare                   0.4.0                 py37_1000    conda-forge
mdtraj                    1.9.2           py37hff88cb2_1000    conda-forge
mmpbsa-py                 16.0                     pypi_0    pypi
mock                      2.0.0                 py37_1001    conda-forge
more-itertools            7.2.0                      py_0    conda-forge
mpiplus                   v0.0.1                py37_1000    conda-forge
msmtools                  1.2.4           py37h3010b51_1000    conda-forge
multiprocess              0.70.7           py37h14c3975_0    conda-forge
natsort                   7.0.1                      py_0    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
netcdf4                   1.5.1.2          py37had58050_0    conda-forge
numexpr                   2.6.9           py37h637b7d7_1000    conda-forge
numpy                     1.16.2           py37h7fc5cc3_0    conda-forge
openblas                  0.3.5             h9ac9557_1001    conda-forge
openmm                    7.4.1           py37_cuda101_rc_1    omnia
openmmtools               0.19.0                   py37_1    omnia
openssl                   1.1.1g               h516909a_0    conda-forge
oset                      0.1.3                      py_1    conda-forge
packaging                 19.2                       py_0    conda-forge
packmol-memgen            1.0.5rc0                 pypi_0    pypi
pandas                    0.24.2           py37hf484d3e_0    conda-forge
parmed                    3.2.0                    pypi_0    pypi
parso                     0.3.4                      py_0    conda-forge
pathos                    0.2.3                      py_0    conda-forge
pbr                       5.1.3                      py_0    conda-forge
pcre                      8.43                 he6710b0_0  
pdb4amber                 1.7.dev0                 pypi_0    pypi
pexpect                   4.6.0                 py37_1000    conda-forge
pickleshare               0.7.5                 py37_1000    conda-forge
pip                       19.0.3                   py37_0    conda-forge
pluggy                    0.13.0                   py37_0    conda-forge
pox                       0.2.5                      py_0    conda-forge
ppft                      1.6.4.9                  py37_0    conda-forge
progress_reporter         2.0                        py_0    conda-forge
progressbar2              3.47.0                     py_0    conda-forge
prompt_toolkit            2.0.9                      py_0    conda-forge
protobuf                  3.7.1            py37he1b5a44_0    conda-forge
psutil                    5.6.1            py37h14c3975_0    conda-forge
pthread-stubs             0.4               h14c3975_1001    conda-forge
ptyprocess                0.6.0                 py37_1000    conda-forge
py                        1.8.0                      py_0    conda-forge
py-xgboost                0.81            py37hf484d3e_1000    conda-forge
pybtex                    0.22.2                   py37_0    conda-forge
pybtex-docutils           0.2.2                    py37_0    conda-forge
pycparser                 2.19                     py37_1    conda-forge
pyemma                    2.5.7            py37hb3f55d8_0    conda-forge
pygments                  2.3.1                      py_0    conda-forge
pymbar                    3.0.5            py37hc1659b7_0    conda-forge
pyopenssl                 19.0.0                   py37_0    conda-forge
pyparsing                 2.3.1                      py_0    conda-forge
pyqt                      5.6.0           py37h13b7fb3_1008    conda-forge
pysocks                   1.6.8                 py37_1002    conda-forge
pytables                  3.5.1            py37h5d69a80_0    conda-forge
pytest                    5.0.1                    py37_0    conda-forge
pytest-runner             5.2                        py_0    conda-forge
python                    3.7.1             h381d211_1003    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
python-utils              2.3.0                      py_1    conda-forge
python_abi                3.7                     1_cp37m    conda-forge
pytraj                    2.0.5                    pypi_0    pypi
pytz                      2018.9                     py_0    conda-forge
pyyaml                    5.1              py37h14c3975_0    conda-forge
qt                        5.6.3                h8bf5577_3  
readline                  7.0               hf8c457e_1001    conda-forge
requests                  2.21.0                py37_1000    conda-forge
sander                    16.0                     pypi_0    pypi
scikit-learn              0.20.3           py37h22eb022_0  
scipy                     1.2.1            py37he2b7bc3_0  
setuptools                40.8.0                   py37_0    conda-forge
sip                       4.18.1          py37hf484d3e_1000    conda-forge
six                       1.12.0                py37_1000    conda-forge
snowballstemmer           2.0.0                      py_0    conda-forge
sphinx                    2.3.1                      py_0    conda-forge
sphinxcontrib-applehelp   1.0.1                      py_0    conda-forge
sphinxcontrib-bibtex      1.0.0                      py_0    conda-forge
sphinxcontrib-devhelp     1.0.1                      py_0    conda-forge
sphinxcontrib-htmlhelp    1.0.2                      py_0    conda-forge
sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
sphinxcontrib-qthelp      1.0.2                      py_0    conda-forge
sphinxcontrib-serializinghtml 1.1.3                      py_0    conda-forge
sqlite                    3.26.0            h67949de_1001    conda-forge
tensorboard               1.13.1                   py37_0    conda-forge
tensorflow-estimator      1.13.0                     py_0  
termcolor                 1.1.0                      py_2    conda-forge
tk                        8.6.9             h84994c4_1000    conda-forge
tornado                   6.0.1            py37h14c3975_0    conda-forge
tqdm                      4.40.0                     py_0    conda-forge
traitlets                 4.3.2                 py37_1000    conda-forge
urllib3                   1.24.2                   py37_0    conda-forge
vampnet                   0.1.4.dev14+g8a86393          pypi_0    pypi
wcwidth                   0.1.7                      py_1    conda-forge
werkzeug                  0.15.2                     py_0    conda-forge
wheel                     0.33.1                   py37_0    conda-forge
xgboost                   0.81            py37hf484d3e_1000    conda-forge
xorg-libxau               1.0.9                h14c3975_0    conda-forge
xorg-libxdmcp             1.1.2             h14c3975_1007    conda-forge
xz                        5.2.4             h14c3975_1001    conda-forge
yaml                      0.1.7             h14c3975_1001    conda-forge
zipp                      0.6.0                      py_0    conda-forge
zlib                      1.2.11            h14c3975_1004    conda-forge
```

**Comment by clonker** [comment]
My best guess is that this has something to do with an unfortunate combination of scipy versions (in particular the packaged lapack) and blas. Perhaps even an incompatible/out of date compilation of a pyemma cython module. However without having something reproducible this is very hard to tell.

**Comment by jaimergp** [comment]
> `scipy                     1.2.1            py37he2b7bc3_0      <blank channel>`

This means `scipy` has been downloaded from `defaults`, and not `conda-forge`. This is a recipe for disaster because ABI compatibility is not guaranteed across most packages in `defaults`. CF's `scipy` is built against different implementations of *BLAS families, like Intel's MKL or OpenBlas. In this case, `defaults` is pulling its own `libopenblas`, built with maybe different library pinnings than Conda Forge.

Anyway, long story short, my first try would be to replace all those packages not coming from `conda-forge` (blank channel column) with `conda-forge` ones.

More info: https://conda-forge.org/docs/user/tipsandtricks.html#using-multiple-channels

**Comment by clonker** [comment]
Thank you for spotting that and providing more insight to the topic!
